packages/itwin/map-layers/src/ui/widget/CustomParamsSettings.tsx

```diff
@@ -0,0 +1,175 @@
+            newParamNames.push(newItem.name);
+            mappingStorage.save(itemKey, {customParamNames: newParamNames});
+          }
+        }
+      }
+    }
+    storage.save(newItem.name, newItem);
+
+    tmpParams[newItem.name] = newItem;
+    setParams(tmpParams);
+    setSelectedValue(undefined); // clear listbox focus
+  }, [mappingStorage, params, storage]);
+
+  const handleAddClick = React.useCallback(() => {
+    UiFramework.dialogs.modal.open(
+      <CustomParamEditDialog
+        onOkResult={onOkEdit}
+        onCancelResult={onCancelEdit}
+      />);
+    return;
+  }, [onCancelEdit, onOkEdit]);
+
+  const onListboxValueChange = React.useCallback((newValue: ListboxValue, _isControlOrCommandPressed?: boolean)=> {
+    const item = params[newValue];
+    if (item)
+      UiFramework.dialogs.modal.open(<CustomParamEditDialog item={item} onOkResult={onOkEdit} onCancelResult={onCancelEdit}/>);
+
+    setSelectedValue(newValue);
+    return;
+  }, [params, onCancelEdit, onOkEdit]);
+
+  return (
+    <div className="customParamsSettings-container">
+      <div className="customParamsSettings-header">
+        <span className="customParamsSettings-header-label">
+          {MapLayersUI.translate("CustomParamSettings.SectionLabel")}
+          <div title={MapLayersUI.translate("Labels.TechPreviewBadgeTooltip")} className="customParamsSettings-previewBadge">
+            <Icon size="small"><SvgTechnicalPreviewMini /></Icon>
+          </div>
+        </span>
+
+        <IconButton size="small" styleType="borderless" className="customParamsSettings-header-add-button" onClick={handleAddClick}>
+          <SvgAdd/>
+        </IconButton>
+      </div>
+      <div className="customParamsSettings-content">
+        <Listbox
+          selectedValue={selectedValue}
+          onListboxValueChange={onListboxValueChange}
+          className="customParamsSettings-content-listbox" >
+          {
+            Object.keys(params).map((keyName) =>
+              <ListboxItem
+                key={keyName}
+                className="customParamsSettings-content-entry"
+                value={keyName}
+                onMouseEnter={() => setListItemUnderCursor(keyName)}
+                onMouseLeave={() => setListItemUnderCursor(undefined)}
+              >
+                <span className="customParamsSettings-content-entry-name" title={keyName}>{keyName}</span>
+                { // Display the delete icon only when the mouse over a specific item otherwise list feels cluttered.
+                  (listItemUnderCursor && listItemUnderCursor === keyName) &&
+                  <>
+                    <IconButton
+                      size="small"
+                      styleType="borderless"
+                      className="map-source-list-entry-button"
+                      title={MapLayersUI.translate("CustomParamSettings.EditButtonTitle")}
+                      onClick={(_event) => {}}>
```

Comment:
Is there a purpose for having an empty `onClick`? Maybe could be deleted otherwise